src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java

```diff
@@ -0,0 +1,174 @@
+                                latch.countDown();
+
+                            }
+                        }
+                    }))
+                    .build();
+
+            long latchTimeout = mongoClientSettings.getSocketSettings().getConnectTimeout(TimeUnit.MILLISECONDS) + 500;
+            MongoClient mongoClient = MongoClients.create(mongoClientSettings);
+
+            try {
+                if (!latch.await(latchTimeout, TimeUnit.MILLISECONDS)) {
+                    throw new Exception("Didn't connect");
+                }
+            } catch (Exception e) {
+                configValue.addErrorMessage("Invalid connection string, unable to reach the server.");
+                mongoClient.close();
+            }
+            if (configValue.errorMessages().isEmpty()) {
+                return Optional.of(mongoClient);
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static void validateHasChangeStreamPermissions(final MongoClient mongoClient, final MongoSourceConfig sourceConfig,
+                                                          final Config config) {
+
+        String database = sourceConfig.getString(MongoSourceConfig.DATABASE_CONFIG);
+        String collection = sourceConfig.getString(MongoSourceConfig.COLLECTION_CONFIG);
+
+        Optional<List<Document>> pipeline = sourceConfig.getPipeline();
+        ChangeStreamIterable<Document> changeStream;
+        if (database.isEmpty()) {
+            changeStream = pipeline.map(mongoClient::watch).orElse(mongoClient.watch());
+        } else if (collection.isEmpty()) {
+            MongoDatabase db = mongoClient.getDatabase(database);
+            changeStream = pipeline.map(db::watch).orElse(db.watch());
+        } else {
+            MongoCollection<Document> coll = mongoClient.getDatabase(database).getCollection(collection);
+            changeStream = pipeline.map(coll::watch).orElse(coll.watch());
+        }
+
+        try (MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor = changeStream.cursor()) {
+            cursor.tryNext();
+        } catch (Exception e) {
+            getConfigByName(config, MongoSourceConfig.CONNECTION_URI_CONFIG).ifPresent(c ->
+                    c.addErrorMessage(format("Unable to create a change stream cursor: %s ", e.getMessage()))
+            );
+        }
+    }
+
+    public static void validateHasReadWritePermissions(final MongoClient mongoClient, final MongoSinkConfig sinkConfig,
+                                                       final Config config) {
+        for (final String topic : sinkConfig.getTopics()) {
+            MongoSinkTopicConfig mongoSinkTopicConfig = sinkConfig.getMongoSinkTopicConfig(topic);
+            MongoNamespace namespace = mongoSinkTopicConfig.getNamespace();
+
+            boolean hasReadWritePermissions = false;
+            try {
+                hasReadWritePermissions = !mongoClient.getDatabase(namespace.getDatabaseName())
```

Comment:
Here I just check if the user has the role.  If we can't check due to lack of perms I defaulted to ignore this check and pass it.